src/storage/redis_metadata.cc

```diff
@@ -497,3 +499,40 @@ rocksdb::Status HyperLogLogMetadata::Decode(Slice *input) {
 
   return rocksdb::Status::OK();
 }
+
+void TDigestMetadata::Encode(std::string *dst) const {
+  Metadata::Encode(dst);
+  PutFixed64(dst, compression);
+  PutFixed64(dst, capacity);
+  PutFixed64(dst, unmerged_nodes);
+  PutFixed64(dst, merged_nodes);
+  PutFixed64(dst, total_weight);
+  PutFixed64(dst, merged_weight);
+  PutDouble(dst, minimum);
+  PutDouble(dst, maximum);
+  PutFixed64(dst, total_observations);
+  PutFixed64(dst, merge_times);
+}
+
+rocksdb::Status TDigestMetadata::Decode(Slice *input) {
+  if (auto s = Metadata::Decode(input); !s.ok()) {
+    return s;
+  }
+
+  if (input->size() < sizeof(uint64_t) * 8 + sizeof(double) + 2) {
```

Comment:
```suggestion
  if (input->size() < sizeof(uint64_t) * 8 + sizeof(double) * 2) {
```

typo?

Reply:
Ahh sorry, my mistake.

Reply:
Sorry that I forget to double check it. 😄

I have updated the statement.

Thanks very much for your warm help!